File: app/model/ListDealsOptions.java
package model;

import java.util.List;
import java.util.Optional;

/**
 * TODO Make fields final so this is immutable
 *
 * */
public class ListDealsOptions {

    public int page;
    public int pageSize;
    public Optional<String> location;
    public Optional<GeoCords> geo;
    public Optional<String> postcode;
    public Optional<List<String>> tags;
    public String sort;
    public String filter;
    public Optional<String> focus;
    public Optional<String> q;

}

File: wowcher-frontend/app/model/ListDealsOptions.scala
package model

/**
 * Created on 22/08/2014.
 */
case class ListDealsOptions
(page: Int, pageSize: Int, location: Option[String], geo: Option[GeoCords],
    postcode: Option[String], tags: Option[List[String]], sort: String,
    filter: String, focus: Option[String], q: Option[String]
     )
